### DISCUSSION TITLE
Search Filter is not showing any results

### DISCUSSION THREAD

**Discussion by JohnSek0609** [microsoft-search/pnp-modern-search#4076] [General]
### Check existing issues before logging new ones
If you are not using the latest version, triage your issue with the latest version before logging as we won't provide patches to older versions.

### Issues related to data mapping, web part setup etc, and SharePoint columns to crawled properties to managed properties should be asked in the discussion section.

**For any question regarding Web Parts usage, please use the ["Discussions"](https://github.com/microsoft-search/pnp-modern-search/discussions) tab instead of creating a new issue.**.

__Remove the above section before submitting, and removal of the below template will result in the issue being closed.__

### Follow the template below or we will have to delete this request!!
We can assist a lot faster if you provide the following information:

**Version used**
Ex: 4.13.1.0

**Describe the bug**
After I properly connecting the Search Filter web part to the Search Results webpart ( vice-versa )  I have added one additional managed property from the "Selected Properties" combobox which I want to be used for filtering. 
This managed property was automatically created as it was part of a content type created in the Tenant Content type gallery.
Even though the property is holding values ( I will add images ) , the search filter web part is not presenting anything.
Additionally I have tried with many other by default added managed properties, like Author, Created etc.. and still the search filter web part is empty, showing nothing to be filtered. There is no error message.

**To Reproduce**
Detailed steps to reproduce the behavior:
Not sure if its needed reproducing, as its very simple setup and maybe its only affecting me.
I have added images of the simple setup which was done at my test site.

**Expected behavior**
Filter values should be shown.

**Screenshots**
![image](https://github.com/user-attachments/assets/292a14de-aecc-4baf-ae1b-1656d26fddfc)
![image](https://github.com/user-attachments/assets/d138b3aa-c024-4680-85c2-b2ab2834db38)
![image](https://github.com/user-attachments/assets/7918c075-dba2-4607-a2b2-18ad4fb7796d)
![image](https://github.com/user-attachments/assets/331df18d-6df6-4bd9-a3f8-ae8ac60d2a9e)
![image](https://github.com/user-attachments/assets/c9541665-f1fe-4ff5-bb24-775490103b1a)
![image](https://github.com/user-attachments/assets/1b19c1cc-240c-41b0-ae04-b098ff869120)

**Desktop (please complete the following information):**

**Additional context**

**Comment by kasperbolarsen** [comment]
_"This managed property was automatically created as it was part of a content type created in the Tenant Content type gallery."_

In order to use a managed property in the search filter it MUST be refinable, which is only the case with a few Out of the box properties, so you will have to map the crawled property to a, in this case, RefinableString, and then add the RefinableString to the Selected Properties AND the Refiner web part 
![image](https://github.com/user-attachments/assets/02f43be6-b736-45ea-9d78-37a1f46e4363)

**Reply by JohnSek0609** [reply]
Hi Kasper, 

Thanks a lot for your quick answer, I really appreciate it.

I understand that, but, I have another question here related to your answer.
Even if I map the crawled property to the Refinable string, how can I use that refinable string for filtering, when the refinable string is not listed in the "Selected Properties" check box, which are coming as part of the returned items by default ( I guess , or )?

Shortly: How can I introduce this new RefinableString managed property which now has the crawled property of that site column ( mentioned earlier ), so I can select first from the "Selected Property"-Search Results webpart and then to use it the Search filter webpart? I guess you know what I mean. Thank you!!

**Reply by kasperbolarsen** [reply]
The properties available for selection in Selected Properties is unfortunately not a complete set, and there is nothing we can do about it. You will have to type in the name of the property in the Selected Properties field

**Reply by JohnSek0609** [reply]
Hi Kasper, thanks a lot, didnt knew that I can manually add a property and it will be recongized.
I have just done that, but still nothing to be filter out from that property.
As you can see by the images, the crawled property ( ows_q_CHCS_KnowledgeBaseCategory )  which was created in the same time with the managed property ( KnowledgeBaseCategoryOWSCHCS ) is now mapped to RefinableString01, but yet nothing shows, do you maybe know why?

![image](https://github.com/user-attachments/assets/89f1b0e8-c4d3-4341-ad19-b991e8950f80)


In the images which I have shown here with the first message it can be seen that if I use the managed property ( KnowledgeBaseCategoryOWSCHCS ) in the search results to get items back. So, that is working, which means the crawled property is not empty , right? Why now, with the RefinableString01 is not ?

Am I missing something else here, do I need to do some site/library re-index, to wait time for the crawler, etc? Thanks!

**Reply by kasperbolarsen** [reply]
Most likely you will just have to wait, but perhaps you should have a look at the troubleshoting section and some of the tools here, as it some great debugging tools, https://microsoft-search.github.io/pnp-modern-search/using-query-tools-to-verify-issues/

**Comment by JohnSek0609** [comment]
Hi Kasper,yes you were right, after some hours I already got data back and I can use it for filtering. 
The only thing which I needed to do additionally  is to refresh the metadata for that site column for the documents which have been added before the setup of the RefinableString01. Do you maybe know why ? Anyway, thank you so much!
